app/src/org/commcare/views/QuestionsView.java

```diff
@@ -375,7 +375,9 @@ public void setOnFocusChangeListener(OnFocusChangeListener l) {
     public void teardownView() {
         for (QuestionWidget widget : widgets) {
             widget.unsetListeners();
+            widget.setOnCreateContextMenuListener(null);
```

Comment:
This was me randomly trying to fix some memory leaks that I was running into early on in this PR.
Worth leaving in because this might cause memory to leak. 
Regardless, Clayton helped me get around the memory leaks by taking an approach that didn't use Views at all when restoring question answers from the saved sessions.